MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionCommitManager.cs

```diff
@@ -33,6 +36,34 @@ public bool ShouldCommitCompletion (IAsyncCompletionSession session, SnapshotPoi
 
 		public CommitResult TryCommit (IAsyncCompletionSession session, ITextBuffer buffer, CompletionItem item, char typedChar, CancellationToken token)
 		{
+			string insertionText;
+			var completionKind = item.Properties.PropertyList[item.Properties.PropertyList.Count - 1].Value;
+			switch (completionKind) {
+			case MSBuildSpecialCommitKind.Element:
+				if (item.InsertText.ToLower () == "packagereference") {
```

Comment:
Why are you special-casing packagereference here?

Reply:
I am ignoring the casing in my new [changes](https://github.com/mhutch/MonoDevelop.MSBuildEditor/pull/19/commits/9e87f3446c0f3eeabc4034676765d2f5d8ded733).

Reply:
I mean, why is this handling PackageReference differently from other elements?

Reply:
After talking to Tom and Jon in my team, we thought that closing tag completion for package reference elements should be like this `<PackageReference Include="Newtonsoft.Json" Version="12.0.2"/>` , instead of `<PackageReference></PackageReference>`. 

Although, I understand that it may be specific to each developer. Someone might want to use the latter one. We can make this special case for `PackageReference ` for other elements too. We can make a list of them and then check if the element is in the list, follow this style of closing tag completion. If you want, I can revert back to general case: `<PackageReference></PackageReference>`

Reply:
I agree with the reasoning but it applies to all items, not just `PackageReference` items. It should also not apply when the commit character is `>`

I would probably suggest adding a new enum value `XmlCompletionItemKind.SelfClosingElement` and using that for item element completion items.

Reply:
Resolved in the following PR: https://github.com/mhutch/MonoDevelop.Xml/pull/1

The changes have been done in the above PR. The above code is no longer there.